aten/src/ATen/cuda/CUDABlas.cpp

```diff
@@ -364,6 +364,7 @@ inline void bgemm_internal_cublaslt(CUDABLAS_BGEMM_ARGTYPES(Dtype)) {
   CuBlasLtMatmulDescriptor computeDesc(computeType, scaleType);
   computeDesc.setAttribute(CUBLASLT_MATMUL_DESC_TRANSA, opa);
   computeDesc.setAttribute(CUBLASLT_MATMUL_DESC_TRANSB, opb);
+  computeDesc.setAttribute<int32_t>(CUBLASLT_MATMUL_DESC_SM_COUNT_TARGET, at::cuda::getCurrentDeviceProperties()->multiProcessorCount - at::globalContext().SMCarveout());
```

Comment:
Somewhat related question: I noticed that when this is unspecified, cuBLAS often does not utilize all the available SMs (it's very common to see 110–120 out of 132 SMs being utilized on H100, maybe except for the stream-k kernels).

- Do you know if the 'under-utilization' was actually for better matmul performance, or was cuBLAS trying to be carve-out conscious?
- Specifying a zero carve-out here means setting CUBLASLT_MATMUL_DESC_SM_COUNT_TARGET to 132 on H100. Would this change the SM utilization of those kernels that would use less than 132 SMs previously?

Reply:
cuBLAS may use thread block cluster sizes that reduce occupancy as an optimization, so it should have nothing to do with carveout if there was no option passed to carveout

"Using larger cluster sizes may reduce the maximum number of active blocks across the GPU (refer to [Occupancy](https://docs.nvidia.com/cuda/hopper-tuning-guide/index.html#sm-occupancy))."
see note on thread block clusters: https://docs.nvidia.com/cuda/hopper-tuning-guide/index.html#thread-block-clusters